src/OrchardCore.Modules/OrchardCore.Contents/Endpoints/Api/CreateEndpoint.cs

```diff
@@ -123,7 +126,7 @@ private static async Task<IResult> HandleAsync(
             await contentManager.SaveDraftAsync(contentItem);
         }
 
-        return TypedResults.Ok(contentItem);
+        return Results.Json(contentItem, options.Value.SerializerOptions);
```

Comment:
To confirm, this is only or content items, right, So we wouldn't use or recommend doing this for non-content item models.

Reply:
I think for most ContentItems this isn't needed. I say that because ContentItem has its own serializer. But I think there are special part that like settings that may need a custom serializer that is in the settings.

Reply:
But to be safe, this could be a general recommendation for all content items unless one knows better, and it wouldn't hurt, right?

Reply:
This is the right approach for any Document not just content item (User, Notification, etc).

I'll look tomorrow to see if we have API that deal with a non-content item.

Reply:
I don't see any other API that return documents. If we get a replay here https://github.com/OrchardCMS/OrchardCore/issues/16831#issuecomment-2395508084 we can confirm that it has been fixed.